### PR TITLE
feat: Change update api to be an instance method

### PR DESCRIPTION
**Pull request by jianlunz-cb**
### What changed? Why?
Change update api from classmethod to instance method
```
>>> sc = SmartContract.register(contract_name="Test Contract", contract_address="0x0000000071727de22e5e9d8baf0edac6f37da039", network_id="base-sepolia", abi=[
...   {
...     "constant": True,
...     "name": "myField",
...     "outputs": [
...       {
...         "name": "",
...         "type": "uint256"
...       }
...     ],
...     "type": "function"
...   }
... ])
CDP API REQUEST: POST https://cloud-api-dev.cbhq.net/platform/v1/networks/base-sepolia/smart_contracts/0x0000000071727de22e5e9d8baf0edac6f37da039/register
CDP API RESPONSE: Status: 200, Data: b'{"abi":"[{\\"constant\\":true,\\"name\\":\\"myField\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"}]","contract_address":"0x0000000071727de22E5E9D8baF0edaC6f37da039","contract_name":"Test Contract","is_external":true,"network_id":"base-sepolia","smart_contract_id":"c9d674b2-ea11-4f7a-b9d2-b7575d05ea80","type":"custom"}\n'
>>> sc.update(contract_name="Test Contract2", abi=[
...   {
...     "constant": False,
...     "name": "myField2",
...     "outputs": [
...       {
...         "name": "",
...         "type": "uint256"
...       }
...     ],
...     "type": "function"
...   }
... ])
CDP API REQUEST: PUT https://cloud-api-dev.cbhq.net/platform/v1/networks/base-sepolia/smart_contracts/0x0000000071727de22E5E9D8baF0edaC6f37da039
CDP API RESPONSE: Status: 200, Data: b'{"abi":"[{\\"constant\\":false,\\"name\\":\\"myField2\\",\\"outputs\\":[{\\"name\\":\\"\\",\\"type\\":\\"uint256\\"}],\\"type\\":\\"function\\"}]","contract_address":"0x0000000071727de22E5E9D8baF0edaC6f37da039","contract_name":"Test Contract2","is_external":true,"network_id":"base-sepolia","smart_contract_id":"c9d674b2-ea11-4f7a-b9d2-b7575d05ea80","type":"custom"}\n'
SmartContract: (smart_contract_id: c9d674b2-ea11-4f7a-b9d2-b7575d05ea80, wallet_id: None, network_id: base-sepolia, contract_address: 0x0000000071727de22E5E9D8baF0edaC6f37da039, type: custom, transaction_hash: None, transaction_link: None, status: None)
```

### CODE REVIEW

**Review by alex-stone — DISMISSED** [review]

**Comment by alex-stone** [comment]
@jianlunz-cb can you update the Changelog.md to include:

* Add support for registering, updating, and listing smart contracts that are
  deployed external to CDP.

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

**Comment by jianlunz-cb** [comment]
> * Add support for registering, updating, and listing smart contracts that are
>   deployed external to CDP.

Fore sure, just updated

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+### [0.13.0] - 2024-12-18
```
Comment (alex-stone) [inline CHANGELOG.md:5]: Let's leave this under Unreleased as we will update this when we actually release!

**Review by alex-stone — APPROVED** [review]

**Pull request merged**